librz/config/config.c

```diff
@@ -386,6 +386,50 @@ static bool __is_true_or_false(const char *s) {
 	return s && (!rz_str_casecmp(s, "true") || !rz_str_casecmp(s, "false"));
 }
 
+RZ_API RzConfigNode *rz_config_set_b(RzConfig *cfg, const char *name, bool value) {
+	RzConfigNode *node = NULL;
+	char *ov = NULL;
+	ut64 oi = 0;
+
+	rz_return_val_if_fail(cfg && cfg->ht, NULL);
+	rz_return_val_if_fail(!IS_NULLSTR(name), NULL);
+
+	node = rz_config_node_get(cfg, name);
+	if (node) {
+		if (rz_config_node_is_ro(node)) {
+			RZ_LOG_DEBUG("(error: '%s' config key is read only)\n", name);
+			return node;
+		}
+
+		ov = node->value;
+		oi = node->i_value;
+		if (rz_config_node_is_bool(node)) {
+			node->i_value = value ? 1 : 0;
+			char *svalue = strdup(rz_str_bool(value));
+			if (svalue) {
+				free(node->value);
+				node->value = svalue;
+			}
+		} else {
+			RZ_LOG_DEBUG("(error: '%s' is not a boolean variable)\n", name);
+			return NULL;
+		}
+	}
+	if (node && node->setter) {
+		if (!node->setter(cfg->user, node)) {
+			if (oi != UT64_MAX) {
```

Comment:
If you took this code from `rz_config_set_i`, it seems that `oi` comes from `node->i_value`. In this function instead it is always 0.

Reply:
It is set in line 418: `oi = node->i_value;`